python/pylibcugraph/pylibcugraph/hits.pyx

```diff
@@ -0,0 +1,209 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Have cython use python 3 syntax
+# cython: language_level = 3
+
+from libc.stdint cimport uintptr_t
+
+from pylibcugraph._cugraph_c.resource_handle cimport (
+    bool_t,
+    data_type_id_t,
+    cugraph_resource_handle_t,
+)
+from pylibcugraph._cugraph_c.error cimport (
+    cugraph_error_code_t,
+    cugraph_error_t,
+)
+from pylibcugraph._cugraph_c.array cimport (
+    cugraph_type_erased_device_array_view_t,
+    cugraph_type_erased_device_array_view_create,
+    cugraph_type_erased_device_array_view_free,
+)
+from pylibcugraph._cugraph_c.graph cimport (
+    cugraph_graph_t,
+)
+from pylibcugraph._cugraph_c.algorithms cimport (
+    cugraph_hits,
+    cugraph_hits_result_t,
+    cugraph_hits_result_get_vertices,
+    cugraph_hits_result_get_hubs,
+    cugraph_hits_result_get_authorities,
+    cugraph_hits_result_free,
+)
+from pylibcugraph.resource_handle cimport (
+    EXPERIMENTAL__ResourceHandle,
+)
+from pylibcugraph.graphs cimport (
+    _GPUGraph,
+)
+from pylibcugraph.utils cimport (
+    assert_success,
+    assert_CAI_type,
+    copy_to_cupy_array,
+    get_c_type_from_numpy_type
+)
+
+
+def EXPERIMENTAL__hits(EXPERIMENTAL__ResourceHandle resource_handle,
+                       _GPUGraph graph,
+                       double tol,
+                       size_t max_iter,
+                       initial_hubs_guess_vertices,
+                       initial_hubs_guess_values,
+                       bool_t normalized,
+                       bool_t do_expensive_check):
+    """
+    Compute HITS hubs and authorities values for each vertex
+
+    The HITS algorithm computes two numbers for a node.  Authorities
+    estimates the node value based on the incoming links.  Hubs estimates
+    the node value based on outgoing links.
+
+    Parameters
+    ----------
+    resource_handle : ResourceHandle
+        Handle to the underlying device resources needed for referencing data
+        and running algorithms.
+
+    graph : MGGraph
```

Comment:
This should actually be both `SGGraph` and `MGGraph`.  We'll have to revisit this for 22.06 and make sure `cugraph.hits` and `cugraph.dask.hits` both call this function with different Graph types.

Reply:
Make sense

Reply:
Fixed